codegenerator/cli/templates/static/codegen/src/eventFetching/hypersync/HyperSync.res

```diff
@@ -121,8 +121,8 @@ module LogsQuery = {
     let missingParams =
       [
         getMissingFields(Types.Log.fieldNames, event.log, ~prefix="log"),
-        getMissingFields(Types.Block.fieldNames, event.block, ~prefix="block"),
-        getMissingFields(Types.Transaction.fieldNames, event.transaction, ~prefix="transaction"),
+        getMissingFields(Types.Block.nonOptionalFieldNames, event.block, ~prefix="block"),
+        getMissingFields(Types.Transaction.nonOptionalFieldNames, event.transaction, ~prefix="transaction"),
```

Comment:
In the future let's replace it with rescript-schema. The con is it'll show only one missing field at a time, but at least will be faster and much less code

Reply:
Ok cool, we do actually have a schema already so we could just use it. You reckon it will be a lot faster?

I also wonder if we should even do this validation. But it caught this bug for us so maybe  it's worth it still.

Reply:
Not as much as the IO usage, but maybe a few milliseconds per batch. Since we already have the schema, might be worth using it here. I'll probably dig into the area, for the Fuel merge anyways, since the have different set of fields we want to include in the field selection.